scipy/sparse/data.py

```diff
@@ -235,27 +243,38 @@ def _arg_min_or_max_axis(self, axis, op, compare):
             indices = mat.indices[p:q]
             am = op(data)
             m = data[am]
-            if compare(m, zero) or q - p == line_size:
-                ret[i] = indices[am]
+
+            if nonzero:
+                if q - p > 0:
+                    ret[i] = indices[am]
+                else:
+                    ret[i] = 0
```

Comment:
`ret` is initialized to all-zeros, so you can drop this else block.

Reply:
Agreed. I fixed this in the latest push.